### ISSUE TITLE
ELECTRA: Training

### ISSUE THREAD

**Issue by MalteHB** [centre-for-humanities-computing/danish-foundation-models#159]
For the training of the ELECTRA model we want to use the HuggingFace `Trainer`. HuggingFace has, however, not yet implemented a fully functional training script of ELECTRA. We, therefore, need to implement our own loss function for our ELECTRA model.
See @richarddwang's implementation [here.](https://github.com/richarddwang/electra_pytorch/blob/4fc0159648bb2bd502f80b1ef6fe320ab302f47c/pretrain.py#L329-L344)

Since @richarddwang uses the `Learner` from `fastai`, we will probably need to extend the HuggingFace `Trainer`, and create our own `ELECTRATrainer` with a `compute_loss` function. Something like the following:
```python
from torch import nn
from transformers import Trainer


class CustomTrainer(Trainer):
    def compute_loss(self, model, inputs, return_outputs=False):
        labels = inputs.get("labels")
        # forward pass
        outputs = model(**inputs)
        logits = outputs.get("logits")
        # compute custom loss (suppose one has 3 labels with different weights)
        loss_fct = nn.CrossEntropyLoss(weight=torch.tensor([1.0, 2.0, 3.0]))
        loss = loss_fct(logits.view(-1, self.model.config.num_labels), labels.view(-1))
        return (loss, outputs) if return_outputs else loss
```

**Comment by MalteHB** [comment]
As we would like to train a DeBERTa V3 in the future, we could look into not simply sharing the generator and discriminator embeddings like the original ELECTRA paper and @richarddwang does [here](https://github.com/richarddwang/electra_pytorch/blob/4fc0159648bb2bd502f80b1ef6fe320ab302f47c/pretrain.py#L366). But rather implementing a "_gradient-disentangled embedding sharing method that avoids the tug-of-war dynamics_".
